testsuite/cli/libra-wallet/src/mnemonic.rs

```diff
@@ -77,7 +77,10 @@ impl Mnemonic {
         bit_writer.write_buffer();
 
         // This will never fail as we've already checked the word-list is not empty.
-        let (checksum, entropy) = bit_writer.bytes.split_last().unwrap();
+        let (checksum, entropy) = bit_writer
+            .bytes
+            .split_last()
+            .unwrap_or_else(|| unreachable!());
```

Comment:
I know @hermanventer really likes this, but -- honestly -- it feels like an anti-pattern that we avoid facets that the language provides... I think expect is okay in "production" code ... unwrap maybe we could agree is bad.

Reply:
Here, this is just a contraction of:
```
match bit_writer.bytes.split_last() {
  Some(tuple) => tuple,
  None => unreachable!()
}
```

except it equips the compiler with the proper kind of information to perform unreachability analysis or use [hardware traps](https://github.com/rust-lang/rust/pull/45920)